assets/src/components/parts/janus-input-number/InputNumber.tsx

```diff
@@ -11,25 +18,93 @@ import { PartComponentProps } from '../types/parts';
 import './InputNumber.scss';
 import { InputNumberModel } from './schema';
 
-const InputNumber: React.FC<PartComponentProps<InputNumberModel>> = (props) => {
-  const [state, setState] = useState<any[]>(Array.isArray(props.state) ? props.state : []);
-  const [model, setModel] = useState<any>(Array.isArray(props.model) ? props.model : {});
-  const [ready, setReady] = useState<boolean>(false);
-  const id: string = props.id;
+/**
+ * If you pass in a string, return the JSON.parse, otherwise return the same value.
+ *
+ * maybeParseJson('{"a": 1}') => {a: 1}
+ * maybeParseJson({"a": 1}) => {a: 1}
+ *
+ */
+const maybeParseJson = (value: any) => {
+  if (typeof value === 'string') {
+    try {
+      return JSON.parse(value);
+    } catch (err) {
+      // bad json, what do?
+      console.error("Couldn't parse JSON to number input", value, err);
+    }
+  }
+
+  return value;
+};
+
+const InputNumber: React.FC<PartComponentProps<InputNumberModel>> = ({
+  id,
+  model: inputModel,
+  notify,
+  onSave,
+  onInit,
+  onReady,
+  onResize,
+}) => {
+  //const debugRef = useRef(props);
+  const model = useMemo(() => maybeParseJson(inputModel), [inputModel]);
```

Comment:
We were using an awkward useState / useEffect combo before for this derived value that would have been causing extra renders and had some weird timing.